src/bindings/html/view.js

```diff
@@ -35,44 +36,30 @@ export class View {
   }
 
   formatValue(id, args) {
-    return this.ctx.formatValue(this.service.languages, id, args);
+    return this.service.languages.then(
+      langs => this.ctx.formatValue(langs, id, args));
   }
 
   formatEntity(id, args) {
-    return this.ctx.formatEntity(this.service.languages, id, args);
+    return this.service.languages.then(
+      langs => this.ctx.formatEntity(langs, id, args));
+  }
+
+  translateFragment(frag) {
+    return this.service.languages.then(
+      langs => translateFragment(this.ctx, this, langs, frag));
   }
 }
 
 View.prototype.setAttributes = setL10nAttributes;
 View.prototype.getAttributes = getL10nAttributes;
-View.prototype.translateFragment = translateFragment;
-
-function onMutations(mutations) {
-  let mutation;
-  let targets = new Set();
-
-  for (var i = 0; i < mutations.length; i++) {
-    mutation = mutations[i];
 
-    if (mutation.type === 'childList') {
-      for (var j = 0; j < mutation.addedNodes.length; j++) {
-        let addedNode = mutation.addedNodes[j];
-        if (addedNode.nodeType === Node.ELEMENT_NODE) {
-          targets.add(addedNode);
-        }
-      }
-    }
-
-    if (mutation.type === 'attributes') {
-      translateElement.call(this, mutation.target);
-    }
-  }
+export function translate(langs) {
+  dispatchEvent(this.doc, 'supportedlanguageschange', langs);
```

Comment:
I'm considering removing `translate` entirely and instead calling `translateDocument` directly from the service, as well as not storing `doc` on `view` and instead keeping track of these relationships in a `WeakMap` in the service.